test(backend): add enhanced integration/unit tests and README coverage

tarun1sisodia opened the pull request:
### Backend Test Suite and Coverage

This backend includes an expanded automated test suite with improved coverage compared to the previous baseline on `main`.

#### Current branch (chore/test-verification)
- Test suites: 11 passed / 11 total
- Tests: 131 total (129 passed, 2 skipped)
- Coverage (all files):
  - Statements: 40.66%
  - Branches: 30.40%
  - Functions: 36.96%
  - Lines: 40.85%

#### Baseline (main)
- Test suites: 5 passed / 5 total
- Tests: 85 total (83 passed, 2 skipped)
- Coverage (all files):
  - Statements: 23.73%
  - Branches: 20.64%
  - Functions: 18.95%
  - Lines: 23.89%

#### Net improvement
- +6 test suites
- +46 tests
- +16.93 pts statements
- +9.76 pts branches
- +18.01 pts functions
- +16.96 pts lines

Notes
- New tests exercise additional routes (`auth`, `dashboard`, `learning`, `health`) and unit layers (`db`, `middleware`, `services`).
- Coverage reports are generated via Jest (`npm test -- --coverage`).

tarun1sisodia committed:
test(backend): add enhanced integration/unit tests and README coverage summary

tarun1sisodia committed:
chore: stop tracking Cursor IDE files and ignore .cursor/

crisecheverria commented:
Hi, can you please address the 4 ESLint errors 

<img width="940" height="618" alt="Screenshot 2025-09-06 at 06 26 47" src="https://github.com/user-attachments/assets/64665cd4-28a4-4d62-b60a-f4b9322119a4" />

tarun1sisodia commented:
GOT it

tarun1sisodia committed:
chore: update .gitignore to ignore additional IDE and rule files; add copilot instructions

tarun1sisodia committed:
feat: add comprehensive GitHub Actions workflow to test entire codebase

- Add full-codebase test workflow with parallel backend/frontend testing
- Include pre-checks to detect changes and optimize CI runs
- Add security audits, coverage reporting, and full integration tests
- Support manual triggering with workflow_dispatch
- Provide comprehensive test summary and status reporting

tarun1sisodia committed:
feat: update GitHub Actions workflow to include package paths for push and pull_request events; add WARP.md to .gitignore

tarun1sisodia committed:
fix: resolve GitHub Actions issues

- Fix rimraf not found error by using npx
- Fix lcov-reporter permission issues with continue-on-error
- Fix React Hook dependency warning using useCallback
- Update path filtering to match backend/frontend workflows
- Improve coverage reporting with better error handling

crisecheverria commented on .cursor/rules/byterover-rules.mdc:
Please remove any commit made to add Cursor files/rules

tarun1sisodia replied:
Yes, I am working on it, will push only tests and resolving the 4 ESLint errors.

crisecheverria commented on .github/copilot-instructions.md:
Please remove any AI instructions file/document.

crisecheverria commented on .github/workflows/test-full-codebase.yml:
No need to add a full codebase test, want to keep them separate for now.

tarun1sisodia replied:
Yes, Got it

crisecheverria commented on packages/backend/README.md:
No need to update Readme with tests status

tarun1sisodia replied:
Okay.

crisecheverria reviewed (commented):
Added some comments to the PR
